src/main/groovy/wooga/gradle/github/base/internal/RepositoryInfo.groovy

```diff
@@ -0,0 +1,38 @@
+package wooga.gradle.github.base.internal
+
+
+import org.ajoberstar.grgit.Grgit
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+
+class RepositoryInfo {
+
+    private static final String DOMAIN = "github.com"
+    private static final String DEFAULT_REMOTE = "origin"
+
+    final Project project
+    final Grgit git
+
+    RepositoryInfo(Project project, Grgit git) {
+        this.project = project
+        this.git = git
+    }
+
+    Provider<String> repositoryNameProviderFromLocalGit() {
+        return project.provider {
+            git.remote.list().find {it.name == DEFAULT_REMOTE}?: null
+        }.map{remote ->
+            def remoteURL = remote.url
+            def domainIndex = remoteURL.indexOf(DOMAIN)
+            def urlAfterDomain = remoteURL.substring(domainIndex + DOMAIN.length() + 1)
+            return urlAfterDomain.replace(".git", "")

```

Comment:
I didn't find any tests for this logic. Specifically for the two main URL types:
* `https://github.com/wooga/atlas-github.git`
* `git@github.com:wooga/atlas-github.git`

From you code it looks like the substring should cut one character after the `github.com` so it should include either the `/` or the `:`. But I would like to have this written down in a test ;)

Reply:
Oh, I just copied this code from somewhere else, and forgot to copy the test. Will put that in place, thanks for the warning.

Reply:
Tests in place in `RepositoryInfoSpec`.